sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -135,67 +156,32 @@ struct _pi_mem : _pi_object {
   // Supporing multi-threaded mapping/unmapping calls
   std::mutex MappingsMutex;
 
-  _pi_mem_type getMemType() const { return MemType; };
+  cm_surface_ptr_t SurfacePtr;
 
 protected:
-  _pi_mem(pi_context ctxt, char *HostPtr, _pi_mem_type MemTypeArg,
+  _pi_mem(pi_context ctxt, char *HostPtr, cm_surface_ptr_t SurfacePtrArg,
```

Comment:
NB: If this object will be deleted through pointer to _pi_object it will leak memory if non-empty Mappings (unordered_map) since _pi_object has non-virtual destructor and this object has a non-trivial one (due to having a member with non-trivial destructor).

Reply:
`virtual ~_pi_object()=default` is applied as we discussed in our weekly meeting.

Reply:
`virtual _pi_object()=default` is applied as we discussed.